hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/s3/tenant/OMTenantRequestHelper.java

```diff
@@ -94,26 +99,26 @@ static void checkTenantExistence(OMMetadataManager omMetadataManager,
    * Retrieve volume name of the tenant.
    */
   static String getTenantVolumeName(OMMetadataManager omMetadataManager,
-      String tenantName) {
+      String tenantId) {
 
     final OmDBTenantInfo tenantInfo;
     try {
-      tenantInfo = omMetadataManager.getTenantStateTable().get(tenantName);
+      tenantInfo = omMetadataManager.getTenantStateTable().get(tenantId);
     } catch (IOException e) {
       throw new RuntimeException("Potential DB error. Unable to retrieve "
-          + "OmDBTenantInfo entry for tenant '" + tenantName + "'.");
+          + "OmDBTenantInfo entry for tenant '" + tenantId + "'.");
     }
 
     if (tenantInfo == null) {
       throw new RuntimeException("Potential DB error or race condition. "
-          + "OmDBTenantInfo entry is missing for tenant '" + tenantName + "'.");
+          + "OmDBTenantInfo entry is missing for tenant '" + tenantId + "'.");
     }
 
-    final String volumeName = tenantInfo.getAccountNamespaceName();
+    final String volumeName = tenantInfo.getBucketNamespaceName();
 
-    if (StringUtils.isEmpty(tenantName)) {
+    if (StringUtils.isEmpty(tenantId)) {
```

Comment:
This should check `volumeName`, not `tenantId`, and should throw an OMException with result code `VOLUME_NOT_FOUND`.

Reply:
Done. but throwing `METADATA_ERROR` instead. The `volumeName` field is not supposed to be null(or "")

Reply:
Good point.

It could also make sense to terminate the OM if the DB is found to be in an undesired state (could be corrupted in some way?) the admin should be notified immediately (i.e. by terminating the OM).

Changed to `VOLUME_NOT_FOUND`.